File: Algorithms/Data Structure dependent/Arrays and String/Sliding Window/findClosestElements.cpp
class Solution {
public:
    vector<int> findClosestElements(vector<int>& arr, int k, int x) {
        auto lo = arr.begin(), hi = arr.end()-1;

        while(hi-lo+1 > k){
            if(std::abs(*lo-x) <= std::abs(*hi-x)) hi--;
            else lo++;
        }

        return std::vector<int>(lo, hi+1);
    }
};

File: Algorithms/Data Structure dependent/Arrays and String/Sorted Arrays/Binary Search/findClosestElements.py
class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
        l, r = 0, len(arr)-k
        while l < r:
            m = (l+r)//2
            # this is our midpoint - now if the value is bigger than our midpoint it's too big
            if x > (arr[m]+arr[m+k])/2 :
                l=m+1
            else:
                r=m
        return arr[l: l+k]